price-feeder/oracle/oracle.go

```diff
@@ -163,14 +163,20 @@ func (o *Oracle) SetPrices() error {
 
 		priceProvider := o.getOrSetProvider(providerName)
 
+		var acceptedPairs []types.CurrencyPair
 		for _, pair := range currencyPairs {
-			if _, ok := requiredRates[pair.Base]; !ok {
-				requiredRates[pair.Base] = struct{}{}
+			if acceptList.Contains(pair.Base) {
+				acceptedPairs = append(acceptedPairs, pair)
+				if _, ok := requiredRates[pair.Base]; !ok {
```

Comment:
Moved the `requiredRates` logic here so that ticks wouldn't fail with unaccepted currency pairs